### DISCUSSION TITLE
Changing a nickname of a member?

### DISCUSSION THREAD

**Discussion by mbround18** [serenity-rs/serenity#2116] [Q&A] [answered]
I play in a D&D server with multiple games going on all the time. I would like to be able to change the nickname of an individual on the fly based on what day of the week it is. Is it possible to change the nickname of an individual?

**Comment by mbround18** [comment, marked as answer]
In case anyone else needs it :) fyi you cant edit the owner of the servers nickname with a bot
```rust
#[command]
async fn command_name(_ctx: &Context, msg: &Message) -> CommandResult {
    let token = env::var("DISCORD_TOKEN").expect("token");
    let http = HttpBuilder::new(token).build();
    let mut payload = JsonMap::new();
    payload.insert(
        "nick".to_string(),
        Value::from("Banana Phone"),
    );

    http.edit_member(
        msg.guild_id.expect("Guild ID is not present!").0,
        msg.author.id.0,
        &payload,
        Some("Its Karl..."),
    )
    .await
    .expect("Failed to set nickname");
    
    Ok(())
}


```

**Reply by justradical** [reply]
> In case anyone else needs it :) fyi you cant edit the owner of the servers nickname with a bot
> 
> ```rust
> #[command]
> async fn command_name(_ctx: &Context, msg: &Message) -> CommandResult {
>     let token = env::var("DISCORD_TOKEN").expect("token");
>     let http = HttpBuilder::new(token).build();
>     let mut payload = JsonMap::new();
>     payload.insert(
>         "nick".to_string(),
>         Value::from("Banana Phone"),
>     );
> 
>     http.edit_member(
>         msg.guild_id.expect("Guild ID is not present!").0,
>         msg.author.id.0,
>         &payload,
>         Some("Its Karl..."),
>     )
>     .await
>     .expect("Failed to set nickname");
>     
>     Ok(())
> }
> ```

thank you so much, I have been looking for this!
